client/src/components/MyAccount/AddItemFormSteps/StepThree.jsx

```diff
@@ -31,6 +38,20 @@ const StepThree = ({
     });
   }, [user]);
 
+  useEffect(() => {
+    (async () => {
+      const response = await getUserCard(user.id);
+      setCardDetails({
+        cardHolderName:
+          response.data.cardHolderName === null
```

Comment:
Also it is always easier to go with expected positive case first, if it isn't null take value, else take empty array.